db/migrate/20200717155424_add_configuration_to_tutorials.rb

```diff
@@ -0,0 +1,5 @@
+class AddConfigurationToTutorials < ActiveRecord::Migration
+  def change
+    add_column :tutorials, :configuration, :jsonb
```

Comment:
This colulmn should have a default value
```suggestion
    add_column :tutorials, :configuration, :jsonb, default: {}
```

FWIW adding columns with default values via a migration can be bad as the operation will lock the table for read / writes. Read more here https://www.braintreepayments.com/blog/safe-operations-for-high-volume-postgresql/

I'm a little surprised our strong migrations gem didn't catch this, https://github.com/ankane/strong_migrations#adding-a-column-with-a-default-value

Reply:
> Instead, add the column without a default value, then change the default.

@camallen's suggestion goes against strong migration's recommendation that you should add a column without a default and then use `change_column_default` to set one. If you want to follow SM's best practices, the suggestion here should be to add a line for that after `add_column`. Are you suggesting to not bother with that because the table is so small?

Reply:
Also! My recent addition to the Users table did both the `add_column` and the `change_column_default` but didn't actually backfill the data. I recommend adding that bit in this PR.

Reply:
Agree with Zach that best practice is to non add a column with a default. 

However for this case with such a small table size we can avoid the extra overhead as the table locking time will most likely be very short due to table size. 
> No need to do any fancy migration here as that table is small (~2K records) so will be milliseconds to add the column + default and no impact to the API.

As this is one of @wgranger early contributions it's up to him if he wants to invest the extra effort required to make this migration align with zero downtime / best practice strategies. Personally i would (to ensure we get into the habit of these) but as the impact risk is very low i've approved his efforts on this PR as it's good enough for the feature addition. 

@zwolf if you feel strongly about enforcing the best practice migration strategies, that's ok too. Please issue a change required review to block my approval review.

Reply:
I see what you're saying. So I think the change here might instead be:

`change_column_default :jsonb, {}`

So the idea would be to rollback, make the change, then rerun the migration?

@camallen does that make sense?

Reply:
yes you will have to rollback your migrated schema to re apply the one you are creating. 

you're updated migration would then look like
``` ruby
add_column :tutorials, :configuration, :jsonb
change_column_default :tutorials, :configuration, {}
```

Noting that if you go this way - you'll also have to backfill the new column see https://github.com/zooniverse/panoptes/pull/3447 
This is due to the fact that the above won't rewrite the table when it adds in the default, rather any new rows will have the default but existing rows won't. 

The way i've suggested above (not best practice) `add_column :tutorials, :configuration, :jsonb, default: {}` would add the default on all column rows during it's rewrite (but it locks the table to do so - ok for ~2k records as it's super quick).

Reply:
Either way is fine with me, I just wanted to be clear about what you were saying here. It's not a lot more code to backfill it (in the same migration) and the strong migrations repo explains how to do it, but I agree that adding the default locking-ly wouldn't be an issue.

Reply:
To be clearer, your options are (approximately, should double check): 

`add_column :tutorials, :configuration, :jsonb, default: {}`
(easiest, not best practice but fine for this)
or

```
add_column :tutorials, :configuration, :jsonb
change_column_default :tutorials, :configuration, {}
Tutorial.in_batches do |tuts|
  tuts.update_all configuration: {}
end
```
(few more steps, but this is how it should be done for bigger tables where a lock would be impactful)

Reply:
Nice. This is pretty clear. Thanks for the descriptive feedback. I think I'll go best practices here to get into good habits.

Reply:
Ended up going the route of adding `default: {}` since this is such a small table. But! The safe migrations gem did kick in and block the migration initially, requiring a `safety_assured` wrapper to be used before migrating. 

Went this route after a chat because of the speed in comparison of how small the table was (~milliseconds of downtime). Noting that in the future, and especially for larger rows, doing this in batches is the preferred method.